triton/kernels/quarot/torch_mm_test.py

```diff
@@ -0,0 +1,113 @@
+import random
+from scipy.linalg import hadamard
+import torch
+import numpy as np
+
+int8_mag_max = 127
+
+# TODO: look into gradient scaling
+
+def quantize(x: torch.HalfTensor):
+    quantize_scale = int8_mag_max / torch.max(torch.abs(x))
+    return (x.type(torch.float64) * quantize_scale).type(torch.int8), quantize_scale.type(torch.float64)
+
+def dequantize(x: torch.CharTensor, quantize_scale):
+    return (x.type(torch.float64) / quantize_scale).type(torch.float16)
+
+def random_rotation_almost_hadamart(size: int, use_hardcoded=False):
+    if use_hardcoded:
+        m = torch.FloatTensor(hadamard(size))
```

Comment:
better to use regular tensor constructor here too and just use dtype parameter

Reply:
Thank you for the feedback. This has been fixed in the most recent commit.